dom/ipc/ContentParent.cpp

```diff
@@ -2795,11 +2805,44 @@ ContentParent::ContentParent(const nsACString& aRemoteType, int32_t aJSPluginID,
                         "XXX(nika): How are we creating a JSPlugin?");
 
   if (RecordAllContentProcesses()) {
+    // When recording all processes we won't be supplied with the dispatch
```

Comment:
Is there a better place for this logic specific to replay that this can be refactored into, instead of putting the logic directly into the `ContentParent` constructor?

Reply:
We don't have a good place for UI process logic like this that is specific to recording/replaying, though I could certainly factor this into a helper function.

Reply:
Yeah even if it's in the same file it'd be nice if that logic was in a place where it was easily moveable to a separate file as needed.